dandi/files/__init__.py

```diff
@@ -127,10 +130,13 @@ def find_dandi_files(
                     # (ie., it's not a Zarr or any other directory asset type
                     # we may add later), so traverse through it as a regular
                     # directory.
-                    if (p / BIDS_DATASET_DESCRIPTION).exists():
+                    if (p / BIDS_DATASET_DESCRIPTION).exists() and not any(
+                        i in p.parents for i in bids_roots
```

Comment:
1. The path queue is extended as part of the while loop, it will always be traversed top-down.
2. Nested BIDS are currently either invalid or derivatives. Support for derivatives is limited in the schema, and absent in the upstream validator. Even if we at some point support derivatives, that will require an object structure redesign. Currently if we validate them, they will be validated as normal BIDS and will fail.

Reply:
IMHO it is ok to fail them if they do not follow the schema of common derivatives which is part of the BIDS. 

- do we have a use case which somehow demands us to pretend that nested BIDS is not BIDS?
- if not -- I would prefer for us to stay consistent and treat BIDS datasets as BIDS datasets, nested they are or not because otherwise it would just then need to treat every file under derivatives as either "I don't care legit " or "illegal" -- situation is nohow better IMHO.